### ISSUE TITLE
Convert task to sub task and vice versa

### ISSUE THREAD

**Issue by generic-user** [johannesjo/super-productivity#235]
Hi,
As an addition to the issue mentioned here: #181 

Could you implement a way of promoting the sub task to task without having to recreate it? 
You could even go the other way round and:  **add tasks** to another **task** as _sub task_ when the task _does not have any sub tasks_ assigned?

## Thanks
Thanks for all the work you put into this repository. It really is a great application. 🎉

**Comment by johannesjo** [comment]
This completely slipped under my radar. I am very sorry. Thanks for the suggestion. This definitely would be nice.

**Comment by santiyounger** [comment]
awesome to see this was already mentioned, would highly aprecciate this feature. This app is excellent, thanks for your hard work

**Comment by santiyounger** [comment]
I just realized an important aspect to consider (when it comes to time tracking)

Since I use super-productivity mostly for time tracking, I realized it would be a lot better for my workflow to  divide tracked time inside project in sub-tasks to create further division of things.

Since all of my tasks already have data in them, currently the only way I can rearrange my exisiting tasks  (with time tracked data) into sub-task is by manually inputting all of the data from different tracked days.

Just something to consider, keeping the time tracked data when converting a task into sub-task.

It would be extremly helpful!

Thank you so much!

**Comment by johannesjo** [comment]
Extracting sub tasks from their parents is no possible via the context menu (or will be with the next release). The other way round still isn't.

**Comment by santiyounger** [comment]
thank you @johannesjo for working on this, this is one of my most needed features, since I love this app and use it all the time.

I tried to like you said "Extract sub tasks from their parents " via drag and drop after updating to version 6.3.3, but I wasn't sure if you meant v6.3.3 or the next release

Thanks!

**Comment by santiyounger** [comment]
> This issue has not received any updates in 90 days. Please comment, if this still relevant!

Hey, super happy with the way to be able to move sub-task from one parent task to another. However I still think it'd amazing to be able to freely move parent tasks into other tasks subtasks.

## Visual Representation
![image](https://user-images.githubusercontent.com/51902730/119367092-3a4a2580-bc77-11eb-8fb4-84663955ba53.png)

![image](https://user-images.githubusercontent.com/51902730/119367159-4afa9b80-bc77-11eb-8ccc-11d36a36ed03.png)

**Comment by JLuisGT** [comment]
+1 to this feature request. :)

**Comment by EngelPika32** [comment]
Still relevant, I suppose : )

**Comment by santiyounger** [comment]
> Still relevant, I suppose : )

definitely!! the lack of this feature definitely made it hard for me to keep use this app for what I needed.  I think it's a great app but this feature would be essential for me to keep using it!

Thanks!

**Comment by nkrot** [comment]
I miss this feature as well.

**Comment by JoseOlin** [comment]
Still relevant and a hard one to find workarounds in my case :)

After brainstorming or defining goals for the week, it is very common that one task becomes a subtask of another or that a task should be divided into smaller pieces.

**Comment by santiyounger** [comment]
@JoseOlin You are right, any workarounds to the lack of this feature become extremely manual and tedious. Hope it can be implemented in the future

**Comment by JoseOlin** [comment]
Still relevant! ;)

**Comment by JoseOlin** [comment]
.

**Comment by JoseOlin** [comment]
.

**Comment by beedell-roke** [comment]
> https://github.com/johannesjo/super-productivity/issues/235#issuecomment-1854976482

@GitHub-Actions, it remains relevant. I'm solely subscribed to issue closure, however.

**Comment by izzortsi** [comment]
Still needed I guess.

**Comment by hkockerbeck** [comment]
Maybe it would be easier to add the basic functionality first and think about refining and polishing it in separate, later stages? The first version could be an additional entry to a task's context menu, similar to what we have for moving a task to a different project. While "Move to project" offers a list of project the task can be moved to, something like "Move as subtask to" could offer a list of potential new parent tasks. 

In most cases, there'll be a higher number of tasks to choose from than projects, so this simple list might get a bit unwieldy. But the functionality would be there. Also, in my experience handling a larger number of tags in the context menu isn't _that_ bad. 

Later stages of refinement could include filtering the list in the context menu (say, with an input field at the top of the list), drag and drop, handling "subtasking" of tasks that have subtasks themselves and so forth.

**Comment by devidfem-beep** [comment]
I would love this feature, perhaps a simple drag to rearrange/re-order task into a sub-task and vice-versa like how TickTick does it would be amazing
Keep up the good work johannesjo!

**Comment by johannesjo** [comment]
@devidfem-beep currently I am waiting for material cdk to fix this, since it probably will be the easiest way to address this.

The problem with this is, that it needs to be rock solid and performant. So I will have very high standards when reviewing this. This being said, I would be happy to review your approach, but it will be much effort and there is a good chance that it will be rejected in the end.

 Please also check the discussion https://github.com/super-productivity/super-productivity/pull/5646

**Comment by johannesjo** [comment]
This will be possible with the next release via drag and drop (if task is not a repeat instance or issue task).
